tools/lib/perlcritic/Perl/Critic/Policy/HashKeyQuotes.pm

```diff
@@ -23,7 +23,7 @@ sub violates {
 
     my $c = $elem->content;
     # special characters
-    return if $c =~ m/[- \/<>.=_:\\\$]/;
+    return if $c =~ m/[-@% \/<>.=_:\\\$]/;
```

Comment:
I would quote the `@` here because one can interpolate arrays in regexes, so for examle `m/[@array]/` would add all elements of `@array`

`m/[-\@% \/<>.=_:\\\$]/;`

Reply:
I don't know what kind of keys we get here, but turning that blacklist check into a whitelist might be better, e.g. only allowing alpha-numeric

Reply:
I see. I wasn't sure about it. Thx